lib/datasource/docker/index.ts

```diff
@@ -113,7 +113,7 @@ async function getAuthHeaders(
     } = hostRules.find({ hostType: 'docker', url: apiCheckUrl });
     opts.json = true;
     if (ecrRegex.test(registry)) {
-      const region = registry.match(ecrRegex)[1];
+      const [, region] = ecrRegex.exec(registry);
```

Comment:
When to use `test` vs `exec`?

Reply:
test simply returns a boolean and does not store capture groups